File: create_qr_code.py
"""
create_qr_code.py
Purpose: Create a QR code for the SENSS poster landing page.
Run this AFTER GitHub Pages gives you the final public website link.

Install first:
    pip install "qrcode[pil]"

Run:
    python create_qr_code.py
"""

import qrcode

POSTER_URL = "https://YOUR-GITHUB-USERNAME.github.io/senss-poster-2026/"

qr = qrcode.QRCode(
    version=None,
    error_correction=qrcode.constants.ERROR_CORRECT_H,
    box_size=20,
    border=4,
)
qr.add_data(POSTER_URL)
qr.make(fit=True)

img = qr.make_image(fill_color="black", back_color="white")
img.save("poster_qr_code.png")

print("QR code saved as poster_qr_code.png")
print(f"QR code points to: {POSTER_URL}")

File: create_qr_code.R
################################################################################
# create_qr_code_clean.R
# Purpose: Generate a clean QR code for SENSS poster
################################################################################

# Install package if needed
if (!requireNamespace("qrcode", quietly = TRUE)) {
  install.packages("qrcode")
}

library(qrcode)

# Replace this with your final GitHub Pages link
poster_url <- "https://williamdormechele.github.io/senss-poster-2026/"

# Create QR code object
qr <- qr_code(poster_url)

# Save as SVG: best for PowerPoint/poster because it stays sharp when resized
generate_svg(qr, filename = "poster_qr_code.svg")

# Optional: also save as PNG preview
png("poster_qr_code.png", width = 1200, height = 1200, res = 300)
plot(qr, col = c("white", "black"))
dev.off()

cat("QR code created successfully:\n")
cat("- poster_qr_code.svg\n")
cat("- poster_qr_code.png\n")